packages/amplify-graphql-model-transformer/src/graphql-model-transformer.ts

```diff
@@ -323,9 +324,30 @@ export class ModelTransformer extends TransformerModelBase implements Transforme
   };
 
   generateResolvers = (context: TransformerContextProvider): void => {
+    const dataSourceMapping: Record<string, string> = {};
     this.resourceGeneratorMap.forEach((generator) => {
       generator.generateResources(context);
+      const ddbDatasources = Object.entries(generator.getDatasourceMap()).filter(
+        ([, datasource]) => datasource.ds.type === 'AMAZON_DYNAMODB',
+      );
+      ddbDatasources.forEach(([modelName, datasource]) => {
+        if (datasource.ds.dynamoDbConfig && !cdk.isResolvableObject(datasource.ds.dynamoDbConfig)) {
+          dataSourceMapping[modelName] = datasource.ds.dynamoDbConfig.tableName;
+        } else {
+          console.warn(
+            `Could not resolve table name for ${modelName}. DataSourceMappingOutput is incomplete. Please manually add ${modelName} to the mapping for your migration.`,
+          );
+        }
```

Comment:
Does this block mean that manually overridden table names aren't properly handled? In what cases would the warning be triggered? Is a warning sufficient? It's super easy to miss warnings in the output of a gen1 flow. Can we allow the customer to specify a manual configuration for the errored configs, and only throw an error if that manual specification isn't present?

Reply:
This was to make TypeScript happy. I haven't found a case where the else would actually execute.

```typescript
Property 'tableName' does not exist on type 'DynamoDBConfigProperty | IResolvable'.
  Property 'tableName' does not exist on type 'IResolvable'.
```

Reply:
How do we handle the cases where table name is changed using `override api`?

Reply:
This will correctly get the table name when using overrides. I just tested again to be sure.

```graphql
type Todo @model {
  id: ID!
  name: String!
  description: String
}
```

Deploy without override adds the correct mapping with the generated table name.

Then I added this override.

```typescript
// set custom table name
resources.models.Todo.modelDDBTable.addPropertyOverride(
  "TableName",
  "MyCustomTableName"
);
```

The table mapping output: `{"Todo":"MyCustomTableName"}`